utils/internal/pre-publish-utils.ts

```diff
@@ -2,27 +2,47 @@
  * Pre-publish utilities to verify that our publish will go smoothly.
  */
 
-const checkPublishConfig = ({name, publishConfig, private: isPrivate}) => {
+const checkPublishConfig = ({
+    name,
+    publishConfig,
+    private: isPrivate,
+    scripts,
+}): boolean => {
+    let returnCode = true;
+
     // first check if is marked as public and there's access to publish the current package
     if (!publishConfig || (!isPrivate && publishConfig.access !== "public")) {
         const requiredAccessType = isPrivate ? "restricted" : "public";
 
         console.error(
             `ERROR: ${name} is missing a "publishConfig": {"access": "${requiredAccessType}"} section.`,
         );
-        process.exit(1);
+        returnCode = false;
     }
 
     // also check if is marked as private and there's restricted access defined
     if (isPrivate && publishConfig.access !== "restricted") {
         console.error(
             `ERROR: ${name} is marked as private but there is a "publishConfig": {"access": "public"} section already defined. Please change it to "access": "restricted" or remove "private": true to make the package public.`,
         );
-        process.exit(1);
+        returnCode = false;
     }
+
+    // check that we are running our pre-publish check for this package
+    if (
+        !scripts.prepublishOnly ||
+        !scripts.prepublishOnly.includes("utils/package-pre-publish-check.sh")
+    ) {
+        console.error(
+            `ERROR: ${name} must have a "prepublishOnly" script that runs "utils/package-pre-publish-check.sh".`,
+        );
+        returnCode = false;
+    }
```

Comment:
**note:** New check to make sure all the packages we publish are running our prepublish check to avoid snapshot releases usurping real releases.